src/DataManager.js

```diff
@@ -14,8 +14,8 @@ function getConfig(url) {
   };
 }
 
-function makeQueryAPICall(currentPage, setData, setErrorMsg, setIsLoading) {
-  const config = getConfig(`https://api.github.com/search/repositories?q=${document.getElementById('searchText').value}&sort=stars&order=desc&per_page=${ITEMS_PER_PAGE}&page=${currentPage}&accept=application/vnd.github.v3+json`);
+function makeQueryAPICall(searchTerm, currentPage, setData, setErrorMsg, setIsLoading) {
+  const config = getConfig(`https://api.github.com/search/repositories?q=${searchTerm}&sort=stars&order=desc&per_page=${ITEMS_PER_PAGE}&page=${currentPage}&accept=application/vnd.github.v3+json`);
```

Comment:
**HIGH-PRI REFACTOR:** DataManager should NOT interact with DOM:
  - document.getElementById('searchText').value should not be in DataManager . Call it in `App`, then pass it as a param to DataManager.
  - Reasons:
    1. Library classes, such as `DataManager,` are meant to be reusable. Having any DOM element interactions there would limit this.
    2. It makes `DataManager` very fragile (`App.js` knows content of the DOM, because it defines the search box, while `DataManager` does not). If we let `DataManager` reads value of the DOM, when `App.js` changes this searchbox, `DataManager` will break immediately.
    3. Easier to test. After this refactor, since `App` & `DataManager` will be no longer coupling, we can write truly isolated unit tests for each.

Reply:
HIGH-PRI REFACTOR:
  - We are passing a lot of setState functions (setData/ErrorMsg/IsLoading) from App to DataManager, making App's internal state coupled with DataManager. This is not good practice.
  - We should make DataManager.makeQueryAPICall more functional (not mutating other's internal state)
  - We can do this by:
    - Not passing any setData/ErrorMsg/IsLoading into DataManager.makeQueryAPICall
    - Instead, make DataManager.makeQueryAPICall return a result, error object. Example:
      {
        errorCode,
        responseData,
      }